draft-ietf-rats-corim.md

```diff
@@ -163,148 +158,129 @@ and {{Section G of -cddl}}.
+~~~ cddl
+ECT = non-empty<{
+  ? &(context: 0) => text
+  ? &(environment: 1) => environment-map
+  ? &(claims: 2) => claims-map / [ + local-ECT ]
+  ? &(authority: 3) => [ + $cryptokeytypechoice ]
+}>
 ~~~
 
-The following describes each member of the `entity-map`.
-
-* `entity-name` (index 0): The name of entity which is responsible for the
-  action(s) as defined by the role. `$entity-name-type-choice` can only be
-  text.  Other specifications can extend the `$entity-name-type-choice` (see
-  {{sec-iana-comid}}).
-
-* `reg-id` (index 1): A URI associated with the organization that owns the
-  entity name
-
-* `role` (index 2): A type choice defining the roles that the entity is
-  claiming.  The role is supplied as a parameter at the time the `entity-map`
-  generic is instantiated.
-
-* `extension-socket`: A CDDL socket used to add new information structures to
-  the `entity-map`.
+Context (label 0):
 
-Examples of how the `entity-map` generic is instantiated can be found in
-{{sec-corim-entity}} and {{sec-comid-entity}}.
+: A text string that TBD (todo: specify here)
 
-### Validity {#sec-common-validity}
+Environment (label 1):
 
-A `validity-map` represents the time interval during which the signer
-warrants that it will maintain information about the status of the signed
-object (e.g., a manifest).
+: A set of claims that identifies instances or classes of environments which are part of a composite Attester's components.
 
-In a `validity-map`, both ends of the interval are encoded as epoch-based
-date/time as per {{Section 3.4.2 of -cbor}}.
+Claims (label 2):
 
-~~~ cddl
-{::include cddl/validity-map.cddl}
-~~~
+: A set of claims with semantics that depend on the context in which an ECT is used.
+In external CoRIM representations, claims can either represent Reference Values, Endorsed Values, or Conditions that have to be met.
+To illustrate examples of internal representations, ECTs are used to represent ACS.
 
-* `not-before` (index 0): the date on which the signed manifest validity period
-  begins
+Authority (label 3):
 
-* `not-after` (index 1): the date on which the signed manifest validity period
-  ends
+: A certain type of key material by which the authority (and corresponding provenance) of an ECT can be determined, such as the public key of an asymmetric key pair that is associated with an authority's PKIX certificate.
 
-### UUID {#sec-common-uuid}
+## Internal Representation {#sec-internal-rep}
 
-Used to tag a byte string as a binary UUID defined in {{Section 4.1.2. of
--uuid}}.
+An ACS is a list of ECTs.
 
 ~~~ cddl
-{::include cddl/uuid.cddl}
-~~~
-
-### UEID {#sec-common-ueid}
-
-Used to tag a byte string as Universal Entity ID Claim (UUID) defined in
-{{Section 4.2.1 of -eat}}.
-
-~~~ cddl
-{::include cddl/ueid.cddl}
+ACS = {
+  &(actual-state: 0) => [ + ECT ]
+}
 ~~~
 
-### OID {#sec-common-oid}
+In this document, CDDL definitions are used to illustrate examples for internal representations.
```

Comment:
Isn't this going a bit far? This document definitely comes with CDDL for CoRIM in CBOR, right?